solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudSolrClient.java

```diff
@@ -101,8 +102,7 @@ public abstract class CloudSolrClient extends SolrClient {
   private final RequestReplicaListTransformerGenerator requestRLTGenerator;
   private final boolean parallelUpdates;
   private ExecutorService threadPool =
-      ExecutorUtil.newMDCAwareCachedThreadPool(
-          new SolrNamedThreadFactory("CloudSolrClient ThreadPool"));
+      Executors.newVirtualThreadPerTaskExecutor();
```

Comment:
What is `ExecutorUtil.newMDCAwareCachedThreadPool()` doing? How does it differ from the API you are using?